### ISSUE TITLE
[$500] [MEDIUM] [Feature Request] Currency selection screen should have "Recents" at top

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#34874]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.29-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @shawnborton 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1705664342497829

## Action Performed:
Go to change the currency on a an expense

## Expected Result: 
We have a list selection pattern where the current selected item is at the top, and we have a section for recents underneath that. I would expect currency selection to work the same.

## Actual Result: 
There is no section for recents, and your currently selected item is not at the top.

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence
![Snip - (2) New Expensify - Google Chrome](https://github.com/Expensify/App/assets/43996225/e67d38df-7352-4f02-ab96-ee26ba685f87)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01964421591b4aea1b</li>
        <li>Upwork Job ID: 1750537225734946816</li>
        <li>Last Price Increase: 2024-01-25</li>
<li>Automatic offers: </li>
<ul>
<li>getusha | Reviewer | 0</li>
<li>DylanDylann | Contributor | 0</li>
</ul></li>
    </ul>
</details>

**Comment by zanyrenney** [comment]
Yep, this looks like it should be external to me. adding the label though!

**Comment by zanyrenney** [comment]
@getusha what do you think of the proposal from @DylanDylann ?

**Comment by getusha** [comment]
@DylanDylann is it possible to implement this future without having to create a new Onyx key?

**Comment by DylanDylann** [comment]
@getusha I think it needs to be stored in BE as well so cannot implement without creating new Onyxkey.

**Comment by getusha** [comment]
Probably adding a new key to the currency object will be ideal?
Pulling internal engineer to take a look at this 🎀 👀 🎀

**Comment by DylanDylann** [comment]
@thienlnam What do you think about this [comment](https://github.com/Expensify/App/issues/34874#issuecomment-1912056684)?

**Comment by thienlnam** [comment]
Yeah, if we're adding recently selected for currency selection we'll likely want another onyx key for consistency with the other recently used items. There's also a few BE changes we'll also need to make to support this update. 

Though instead of keying this by policy, we should probably just create a general recentlyUsedCurrencies field for the account - currency selection is typically user-centric and independent of individual policy constraints

**Comment by thienlnam** [comment]
@zanyrenney Could you see which wave this would fit in to see if we should work on this now or not?

**Comment by zanyrenney** [comment]
sure thing! i think this should be in wave6 but I will drop it in for greg to make a call.

**Comment by zanyrenney** [comment]
https://expensify.slack.com/archives/C02MW39LT9N/p1706530608517779

**Comment by getusha** [comment]
We can proceed with @DylanDylann's proposal, i think we should first complete the backend changes first @thienlnam?

**Comment by thienlnam** [comment]
Yup, we'll need to do that first - let me see if someone else is interested is taking these changes on

**Comment by zanyrenney** [comment]
find anyone interested in taking on the changes @thienlnam

**Comment by zanyrenney** [comment]
?

**Comment by thienlnam** [comment]
You're looking at him! 😆 I'll get the internal changes up later this week

**Comment by zanyrenney** [comment]
woo thanks @thienlnam please update the issue with your progress 🎉

**Comment by thienlnam** [comment]
Got started on a draft PR - looks like we haven't been storing this anywhere since we always rely on the user's current location

Currently going through all the places that a currency could be used - scan / create / update and then adding tests

**Comment by zanyrenney** [comment]
Nice, thanks for the update @thienlnam can you let me know how your progress is going now?

**Comment by zanyrenney** [comment]
Bump @thienlnam how you going here?

**Comment by thienlnam** [comment]
Haven't made any new updates here - been working on Track Expense. Will likely get back to this next week

**Comment by zanyrenney** [comment]
waiting on @thienlnam - changing to make him owner until its able to be prioritised as there is nothing I or @getusha can do for now until then.

**Comment by getusha** [comment]
@thienlnam is OOO, will get back to this when he returns.

**Comment by zanyrenney** [comment]
Not sure where you got that from @getusha - i checked internally @thienlnam isn't OOO yesterday or day before 😅

**Comment by zanyrenney** [comment]
@thienlnam can you please update this issue with your work / what the latest update is.

**Comment by thienlnam** [comment]
I was OOO on Monday for President's day. I'm back now, I've been catching up and focused on Track which is blocking some other wave 6 critical issues so I haven't made any new progress.

I don't think this is that urgent as a new feature, so I'm going to move this to a monthly - though feel free to find someone to take it over if you would like it done sooner

**Comment by getusha** [comment]
>Not sure where you got that from @getusha 

@zanyrenney noticed it from slack profile :)

**Comment by zanyrenney** [comment]
oh weird haha i didn't see that in our internal update for OOO, but all good, thanks!

**Comment by thienlnam** [comment]
BE PR is merged so we're saving all the used currencies by policy, but we're not sending any pusher updates - that's currently held on Ioni's [project](https://github.com/Expensify/Expensify/issues/364560) of `Standardize how we save NVPs in onyx`.

Going to update the issue title to reflect that

**Comment by zanyrenney** [comment]
On hold for issue linked in title.

**Comment by zanyrenney** [comment]
On HOLD.

**Comment by zanyrenney** [comment]
Polish / waiting for BE changes.

**Comment by zanyrenney** [comment]
hey @thienlnam when do you think we'll be prioritising this one?

**Comment by thienlnam** [comment]
@iwiznia Seems like we're pretty close to sending out all account updates, when do you think that will be live?

**Comment by iwiznia** [comment]
> close to sending out all account updates
I don't know what you mean

**Comment by thienlnam** [comment]
For this project https://github.com/Expensify/Expensify/issues/364560

**Comment by zanyrenney** [comment]
Any update further here please @thienlnam

**Comment by thienlnam** [comment]
Ah yeah, so the BE changes are done - but we didn't send any pusher updates in anticipation for this [change](https://github.com/Expensify/Expensify/issues/364560) which will send account NVPs automatically.

Once that is done, we can turn this external to have someone add the recents key

**Comment by iwiznia** [comment]
That project is done, I even sent the project done email today./

**Comment by thienlnam** [comment]
Oh nice! Okay in that case I just need to pass the onyx key to the client (doing that [here](https://github.com/Expensify/Auth/pull/10597)), and then a contributor can start on the FE changes

**Comment by DylanDylann** [comment]
@thienlnam Could you check [this comment](https://github.com/Expensify/App/issues/34874#issuecomment-1918776952)? I can handle this one

**Comment by zanyrenney** [comment]
bump @thienlnam please can you respond to the comment above?

**Comment by thienlnam** [comment]
Yup, once the Auth PR is merged you can go ahead and handle the FE changes. 

The onyx key will be under nvp_recentlyUsedCurrencies

**Comment by thienlnam** [comment]
@DylanDylann You should be ready to start on this now - the holding code is deployed

The currencies are keyed by policyID

```
nvp_recentlyUsedCurrencies: {
   01DC7E46150451C4: ['USD', 'CAD'],
   0426477218E500C1: ['USD', 'CAD'],
   1CAFAAA6EAD09539: ['USD', 'CAD'],
}
```

**Comment by DylanDylann** [comment]
@thienlnam I am working on this issue but I cannot find the ```nvp_recentlyUsedCurrencies```. Can you help check whether the BE changes are deployed?

**Comment by thienlnam** [comment]
You might need to create / edit some transactions to populate it so that it is returned - let me know if that works

**Comment by DylanDylann** [comment]
Yes. I tried to create a transaction, but there is no ```nvp_recentlyUsedCurrencies``` data returned
![image](https://github.com/Expensify/App/assets/141406735/8ec87672-6de1-490a-900e-b2d464933d8c)

**Comment by thienlnam** [comment]
Ah okay, it seems like there are still a few locations where the currency needs to be updated. 

Right now this works if you go into Expensify classic, create a transaction, add it to a report on a workspace, and then change the currency.

That will get the NVP to show up in Onyx. Anyways, the structure on the onyx key won't change so you should be able to start with that and use Expensify classic to load the initial test data

**Comment by DylanDylann** [comment]
@getusha @thienlnam PR https://github.com/Expensify/App/pull/41834 is ready

**Comment by thienlnam** [comment]
We're still working through this

**Comment by thienlnam** [comment]
Latest is here: https://expensify.slack.com/archives/C02NK2DQWUX/p1725039240080049?thread_ts=1724690852.626429&cid=C02NK2DQWUX

Currently pending a PR in the BE to change the format, and then an update on the App PR

**Comment by DylanDylann** [comment]
@zanyrenney Please help to process payment here

**Comment by zanyrenney** [comment]
**payment summary**

$500 paid to Duong via Upwork.
$50o owed to @getusha who is paid via ND.

**Comment by garrettmknight** [comment]
$500 approved for @getusha
